RELEASING.md

```diff
@@ -9,8 +9,12 @@
       (otherwise the workflow will pick up the version from `main` and just remove the `.dev` suffix).
   * Review the two pull requests that it creates.
     (one is targeted to the release branch and one is targeted to `main`).
-    * Merge the one targeted towards the release branch.
-    * The builds will fail for the `main` pr because of validation rules. Follow the [release workflow](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/RELEASING.md) for the contrib repo up until this same point. Change the SHAs of each PR to point at each other to get the `main` builds to pass.
+    * The builds will fail for both the `main` and release pr because of validation rules. Follow the [release workflow](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/RELEASING.md) for the contrib repo up until this same point.
+    * Change the CONTRIB_REPO_SHA of core PRs to point to their counterpart in contrib.
+    * Change the CORE_REPO_SHA of contrib PRs to point to their counterpart in core.
```

Comment:
Discussed offline, we could just set the release branch name in the "Create release branch" workflow. Then they never have to change. It's not perfect because it's a "moving target" vs SHA but less toilsome

Reply:
I'm OK to address this later and get this PR in